text/0000-bitfields.md

```diff
@@ -0,0 +1,340 @@
+- Feature Name: bitfield
+- Start Date: 2016-01-07
+- RFC PR: (leave this empty)
+- Rust Issue: (leave this empty)
+
+# Summary
+[summary]: #summary
+
+Add support for bitfields via a single per-field attribute.
+
+# Motivation
+[motivation]: #motivation
+
+Bitfields are commonly encountered when interacting with C code. Since many of
+the details of the behavior of bitfields are left to the implementation, it is
+hard to write cross-platform rust code that uses bitfields correctly.
+
+Currently, such code often manually collapses bitfields into a single integer
+field and attempts to extract and set the individual fields via bitshifts. This
+approach works for simple bitfields and when the full set of target platforms is
+known, however, once the bitfields become more complicated or if the code is
+compiled for an unexpected target, such code will silently fail. In particular,
+code that assumes that bitfields are placed into the integer in increasing bit
+order will likely break on big-endian machines.
+
+Appendix A contains various examples of the (to some) surprising behavior of
+bitfields on the x86-64 linux platform.
+
+This rfc introduces a single attribute that can be applied to fields inside of
+`#[repr(C)]` structs to turn them into bitfields.
+
+# Detailed design
+[design]: #detailed-design
+
+The attribute `#[bitfield(N)]` is added. Here, `N` can be any constant
+expression that evaluates to a value of type `usize`.
```

Comment:
FWIW, I'd like the have a solution which can be later extended to support "repr(Rust)" bitfields with specified and standardized ABI (not "implementation defined"), FFI is certainly not the only application of bitfields.
If constant expressions in attributes are indeed implemented, then this seems to be a good enough for native Rust bitfields as well. (Maybe there are better ways, I don't know)